packages/cosmic-swingset/src/extract-proposal.js

```diff
@@ -97,25 +96,60 @@ export const extractCoreProposalBundles = async (
         );
         return handle;
       };
-      const proposal = await deeplyFulfilled(
-        harden(ns.defaultProposalBuilder({ publishRef, install })),
-      );
-      const { sourceSpec, getManifestCall } = proposal;
+      const proposal = await ns.defaultProposalBuilder({ publishRef, install });
+
+      const bundleSpecs = [];
+      // Add the proposal bundle handles in sorted order.
+      [...bundleHandles.values()]
+        .sort(({ source: a }, { source: b }) => {
+          if (a < b) {
+            return -1;
+          }
+          if (a > b) {
+            return 1;
+          }
+          return 0;
+        })
+        .forEach((handle, j) => {
+          handle.bundleID = `coreProposal${i}_${j}`;
+          harden(handle);
+          bundleSpecs.push([handle.bundleID, { sourceSpec: handle.source }]);
+        });
```

Comment:
Why not do a `.map` ?

```suggestion
      // Add the proposal bundle handles in sorted order.
      const bundleSpecs = [...bundleHandles.values()]
        .sort(({ source: a }, { source: b }) => {
          if (a < b) {
            return -1;
          }
          if (a > b) {
            return 1;
          }
          return 0;
        })
        .map((handle, j) => {
          handle.bundleID = `coreProposal${i}_${j}`;
          harden(handle);
          return [handle.bundleID, { sourceSpec: handle.source }];
        });
```